examples/gno.land/r/leon/hof/datasource.gno

```diff
@@ -0,0 +1,64 @@
+package hof
+
+import (
+	"errors"
+
+	"gno.land/p/demo/seqid"
+	"gno.land/p/demo/ufmt"
+	"gno.land/p/gnome/datasource"
```

Comment:
@leohhhn this package is the one that would cause issues with the way `hof` realm works. I'm not really sure how to deal with it. I think at this point we would need such a package to be available in the main repo.

I could try to come up with an abstraction generic enough for this case, based on the `gno.land/p/gnome/datasource` package. Maybe under `gno.land/p/demo/datasource` for example.

Reply:
I think it's perfectly fine to have it under p/gnome/datasource, or, if you want to keep the `gnome` namespace clean for now, you can add it to your own personal namespace, ie `p/jeronimoalbi`.

Reply:
Going to do that then, makes sense

Reply:
@leohhhn, I just created #3318 with a generic datasource implementation. If it's merged then I will update this PR to use the new `datasource` package.